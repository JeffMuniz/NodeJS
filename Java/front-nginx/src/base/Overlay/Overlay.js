import React, { Component } from "react";
import { node, func } from "prop-types";
import { Container } from "./Overlay.styles";

class Overlay extends Component {
  componentDidMount() {
    document.body.style.overflow = "hidden";
  }

  componentWillUnmount() {
    document.body.style.overflow = null;
  }

  render() {
    const { onClick, children } = this.props;

    return (
      <Container id="overlay" onClick={onClick}>
        {children}
      </Container>
    );
  }
}

Overlay.propTypes = {
  children: node.isRequired,
  onClick: func,
};

Overlay.defaultProps = {
  onClick: () => null,
};

export default Overlay;
