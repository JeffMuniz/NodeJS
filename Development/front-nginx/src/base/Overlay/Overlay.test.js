import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Overlay from "./Overlay";

const defaultProps = {
  children: <div>test</div>,
  handleClick: jest.fn(),
};

describe("Voucher Section - Unit Test", () => {
  it("Should render Voucher Section type food", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<Overlay {...defaultProps} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
