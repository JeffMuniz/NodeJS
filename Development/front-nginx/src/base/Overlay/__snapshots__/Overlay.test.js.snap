// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voucher Section - Unit Test Should render Voucher Section type food 1`] = `
<styled.div
  id="overlay"
  onClick={[Function]}
>
  <div>
    test
  </div>
</styled.div>
`;
